sdk/engine/source/graphics/AssetsStorage.cpp

```diff
@@ -0,0 +1,65 @@
+#include "assets/Assets.h"
+#include "graphics/AssetsStorage.h"
+#include "graphics/Base.h"
+#include "graphics/GpuAllocator.h"
+
+namespace nc::graphics
+{   
+    AssetsStorage::AssetsStorage(Base* base, GpuAllocator* allocator, nc::GpuAccessorChannels* gpuAccessorChannels)
+        : m_base{base},
+          m_allocator{allocator},
+          m_vertexData{m_base, m_allocator},
+          m_indexData{m_base, m_allocator},
+          m_gpuAccessorChannels{gpuAccessorChannels},
+          m_onMeshAddConnection{m_gpuAccessorChannels->onMeshAdd->Connect(this, &AssetsStorage::UpdateMeshBuffer)}
+    {
+    }
+
+    const VertexData& AssetsStorage::GetVertexData() const noexcept
+    {
+        return m_vertexData;
+    }
+
+    const IndexData& AssetsStorage::GetIndexData() const noexcept
+    {
+        return m_indexData;
+    }
+    
+    void AssetsStorage::SetVertexData(const std::vector<graphics::Vertex>& vertices)
+    {
+        m_vertexData.buffer.Clear();
+        m_vertexData.buffer = ImmutableBuffer(m_base, m_allocator, vertices);
+    }
+
+    void AssetsStorage::SetIndexData(const std::vector<uint32_t>& indices)
+    {        
+        m_indexData.buffer.Clear();
+        m_indexData.buffer = ImmutableBuffer(m_base, m_allocator, indices);
+    }
+
+    void AssetsStorage::UpdateMeshBuffer(const MeshAsset& meshAsset)	
+    {	
+        SetVertexData(meshAsset.vertices);	
+        SetIndexData(meshAsset.indices);	
+    }
+
+    VertexData::VertexData(Base* base, GpuAllocator* allocator)
```

Comment:
Use a member initializer list here and in `IndexDara`. Also simply `Vertex{}` should give you a zeroed out struct.